export_snowflake/__init__.py

```diff
@@ -522,26 +528,126 @@ def flush_records(stream: str,
     db_sync.delete_from_stage(stream, s3_key)
 
 
+def transfer_data_in_parallel(s3_csv_lists, config, o, file_format_type):
+    parallel_transfer_start_time = time.time()
+    
+    # make the decision of how many processes to use later
+    pool = multiprocessing.Pool(4, initialize_s3_client_db_sync(config, o, file_format_type))
+    pool.map(download_file, s3_csv_lists)
+    pool.close()
+    pool.join()
+    
+    parallel_transfer_end_time = time.time()
+    stream = config["stream"]
+    LOGGER.info(f"Start time for transfering {stream} as {parallel_transfer_start_time} and end time as {parallel_transfer_end_time}")
+
+
+def download_file(job):
+    bucket, key, filename, stream = job
+
+    s3_client.download_file(bucket, key, filename)
+
+    # upload to snowflake
+    file_path = filename.replace("\\", "/")
+
+    s3_key = db_sync.put_to_stage(file_path, stream, temp_dir=None)
```

Comment:
could there be chance of name clash for files in snowflake stage? Its good idea to add Symon specific naming to filenames to distinguish for exports running from Symon.
Is there a chance for `run` to generate same file names for different exports of same pipe? Add export task id in naming?

Reply:
I think there shouldn't be name collision issue here as we are using the file names from the generated export results but yea it's good idea to add the specific name to tell the difference. I will append the `export task id` to it.

Reply:
Do we have a case where special characters are included in the filename? If we do, might want to double check snowflake id requirements - i remember some special characters are not allowed without wrapping double quotes, but mot sure if it will affect put command. 

For snowflake import using copy command into user stage, I ended up copying the files into uuid4 (random numbers) generated subfolder in user stage.

Reply:
shouldn't be special characters in the export bucket file list. should just be the file names like `part-0000-1d2eb8b0-304d-4680-bf37-77864cb39fff-c000.csv.gz` and schema.json files.

Reply:
Also, we might want to consider using user stage in the future as enhancement, but would need design decision.
- Table stage requires ownership on the table and there could be only 1 owner. This allows altering primary key of the table, but different user cannot export to that table since it will get privilege error while writing to the table stage. (current behaviour)
- User stage exists per user so different user can export to same table, but not sure if altering primary key is supported if the user is missing ownership on the table.

Reply:
I will log this in the confluence page, sounds like a potential behaviour change for different user to export to the same table. Guess we will need PM decision for it.
Don't think any technical blocks for this, we can create stage with Snowflake query syntax and remove afterwards.